floris/heterogeneous_map.py

```diff
@@ -50,20 +53,40 @@ def __init__(
         if not isinstance(speed_multipliers, (list, np.ndarray)):
             raise TypeError("speed_multipliers must be a numpy array or list")
 
+        # If z is provided, check that it is a list or numpy array
+        if z is not None:
+            if not isinstance(z, (list, np.ndarray)):
+                raise TypeError("z must be a numpy array or list")
+
         # Save the values
         self.x = np.array(x)
         self.y = np.array(y)
         self.speed_multipliers = np.array(speed_multipliers)
 
+        # If z is provided, save it as an np array
+        if z is not None:
+            self.z = np.array(z)
+        else:
+            self.z = None
```

Comment:
This is fine, but might be cleaner to have a single liner
`self.z = z if z is None else np.array(z)`

Reply:
I feel like I'm arguing every point, and I don't mean to, appreciate the suggestions!  I just somehow prefer the multi-line